saleor/order/events.py

```diff
@@ -469,3 +474,154 @@ def order_note_added_event(*, order: Order, user: UserType, message: str) -> Ord
         parameters={"message": message},
         **kwargs,
     )
+
+
+def _prepare_discount_object(
+    order_discount: "OrderDiscount",
+    old_order_discount: Optional["OrderDiscount"] = None,
+):
+    discount_parameters = {
+        "value": order_discount.value,
+        "amount_value": order_discount.amount_value,
+        "currency": order_discount.currency,
+        "value_type": order_discount.value_type,
+        "reason": order_discount.reason,
+    }
+    if old_order_discount:
+        discount_parameters["old_value"] = old_order_discount.value
+        discount_parameters["old_value_type"] = old_order_discount.value_type
+        discount_parameters["old_amount_value"] = old_order_discount.amount_value
+    return discount_parameters
+
+
+def order_discount_event(
```

Comment:
I would consider moving these to a new file `saleor/order/events/discounts.py` but if you don't have time let's skip it for now. Later we could try to refactor other events.